app/controllers/subscriptions_controller.rb

```diff
@@ -0,0 +1,54 @@
+class SubscriptionsController < ApplicationController
+  before_filter :authorize
+  before_action :set_stripe_customer, only: [:show, :update, :destroy]
+
+  def show
+  end
+
+  def create
+    customer = Stripe::Customer.create(
+      source: params[:stripeToken],
+      plan: "ca-basic",
+      email: current_user.email
+    )
+
+    resource.update(
+      stripe_id: customer.id
+    )
+
+    redirect_to subscription_path
+  end
+
+  def update
+    customer = @stripe_customer
+
+    card_id = customer.sources.data[0].id 
+
+    customer.sources.retrieve(card_id).delete() 
+
+    customer.sources.create(
+      source: params[:stripeToken]
+    ) 
+
+    redirect_to subscription_path
+  end
+
+  def destroy
+    customer = @stripe_customer
+    card_id = customer.sources.data[0].id 
+    customer.sources.retrieve(card_id).delete()
```

Comment:
so we should make a `delete_stripe_card` method or something like that? will do, after making the tests